src/schemas/FEEDBACK-TOOL/schema.js

```diff
@@ -92,7 +92,50 @@ const internationalSchoolAddress = {
   }
 };
 
-const schoolAddress = [domesticSchoolAddress, internationalSchoolAddress];
+// TRANSLATE: all search tool address fields into street, street2, and country
+const searchToolSchoolAddress = {
+  required: ['viaSearchTool', 'street', 'city', 'country'],
+  properties: {
+    viaSearchTool: {
```

Comment:
This is actually not necessary to specify, looping in @jholton to clarify if possible. As long as the data satisfies one of the schemas, it will be considered valid, it is not necessary to specify which schema should be used for the validation.
Also, if we plan to include information in the schema that is not intended to be sent back (besides using `view:` fields on the frontend), sometimes it's helpful to add a comment in the schema, to call out that we're proposing the backend perform a translation, to remove that information.

Reply:
I think we actually do need it. Imagine a case where the country is 'United States' but we don't pass in all the data that's required for a US address, such as `postalCode`. Then it will fail the validation for a domestic address but still pass the new validation I just added. So without adding this additional flag we won't be able to enforce the additional requirements for US addresses.

Reply:
We will need to specify which domestic address schema we are using on the frontend, so there is no risk of the wrong domestic schema being used to validate a particular field.

Reply:
How would we do that unless the schemas have different shapes? That's what I was trying to accomplish with adding `viaSearchTool`.

Reply:
Specific types of schemas are selected from oneOf lists on the frontend as:
```
const domesticSchoolAddress = schoolAddress.oneOf[0];
const internationalSchoolAddress = schoolAddress.oneOf[1];
```
https://github.com/department-of-veterans-affairs/vets-website/blob/b93fdd14b3dcaea712a3b3aade48499190d666a5/src/applications/edu-benefits/feedback-tool/config/form.js#L40-L41

Reply:
Okay, I need to process this a bit.

Reply:
@cehsu Other than prevent invalid input on form fields, is there another layer of frontend validation? I'm asking because picking a school via the search tool bypasses all of the form input validation.

Reply:
incorrectly prefilled/fetched data prevents users from advancing to the next page (e.g. the service branch, and contact info address country bugs) on the frontend, and prevents submission to the backend, i believe, @jholton does that sound correct?

Reply:
Yes, if the submitted form does not pass schema validation on the backend, the api will raise an exception and return a 422.

Reply:
@cehsu After looking at this a lot more, I'm still thinking the method I came up with is a valid choice, but I have some questions about what the existing code is still doing. First, I noticed that the two school address schemas that are brought in are used, but all of their [required attributes are stripped out](https://github.com/department-of-veterans-affairs/vets-website/blob/search-tool-school-address/src/applications/edu-benefits/feedback-tool/config/form.js#L44) so all the validation is done via `ui:required`s. I'm not sure why that is. ~~I'm also not sure why the schema for school address is [updated here](https://github.com/department-of-veterans-affairs/vets-website/blob/search-tool-school-address/src/applications/edu-benefits/feedback-tool/config/form.js#L409) if all of the form fields stay the same whether it's a domestic or international address.~~ (I think I figured that one out. It has to do with some of the string lengths being different for domestic vs intl)

When using the school search tool, I'm not sure we even can do validation on the front end since the tool is simply setting the school address data directly. I still need to look into that a bit more.

Reply:
Good point, a user shouldn't encounter a validation error using the school search tool. Is it possible to determine that the data coming in will validate properly? Should we set up error logging in case we encounter fields that are too long? Should we also create an abbreviation helper for strings that are too long (e.g. replace street with ST).

Reply:
I had a similar thought after submitting this. I think that abbreviation helpers would be a bit heavy-handed. I'm assuming that the backend will completely reject the submission if a field is too long and thus fails the JSON schema validation check. If that's the case, I think we should just truncate fields that are too long. Even if the entire address is not captured in Salesforce, having the fist ~53 characters in each street field should be more than enough info for someone to know which institution is being reported.